rdm/util.py

```diff
@@ -93,3 +94,44 @@ def and_list_str(items):
         return items[0] + ' and ' + items[1]
     else:
         return ', '.join(items[:-2] + [and_list_str(items[-2:])])
+
+
+def use_auto_section_numbering(context):
+    filter_specifcation_list = context.get('system', {}).get('post_filters', [])
+    return 'auto_section_numbers' in filter_specifcation_list
+
+
+def empty_formatter(spacing, tag, content):
+    return ''
+
+
+def plain_formatter(spacing, tag, content):
+    return '{spacing}[{tag}{content}]'.format(spacing=spacing, tag=tag, content=content)
+
+
+def create_formatter_with_string(format_string):
+    def custom_formatter(spacing, tag, content):
+        return format_string.format(spacing=spacing, tag=tag, content=content)
+
+    return custom_formatter
+
+
+def dynamic_class_loader(extension_descriptor_list):
+    result = []
+    for extension_descriptor in extension_descriptor_list:
+        module_name, class_name = extract_module_and_class(extension_descriptor)
+        module = import_module(module_name)
+        class_object = getattr(module, class_name)
+        result.append(class_object)
+    return result
+
+
+def extract_module_and_class(descriptor):
+    parts = descriptor.split('.')
+    module_name = '.'.join(parts[:-1])
+    class_name = parts[-1]
+    return module_name, class_name
+
+
+def post_processing_filter_list(environment):
```

Comment:
This function doesn't seem like a generic utility.  Perhaps it should go in `render.py`?

Reply:
It is used by both `render` and by `RdmExtension`. Moving it to `util` allows use by both.